src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -146,24 +160,44 @@ private void CleanupTasksAndRegistrations(long taskId)
             }
         }
 
+        /// <summary>
+        /// Serialize the args to a json string, with the byte arrays
+        /// extracted out for seperate transmission.
+        /// </summary>
+        /// <param name="args">Argument(s) to be converted to json.</param>
+        /// <returns>
+        /// A tuple of the json string and an array containing the extracted
+        /// byte arrays from the args.
+        /// </returns>
+        protected internal (string, byte[][]?) SerializeArgs(object? args)
+        {
+            ByteArraysToSerialize.Clear();
+            var serializedArgs = JsonSerializer.Serialize(args, JsonSerializerOptions);
+            var byteArrays = ByteArraysToSerialize.ToArray();
+
+            return (serializedArgs, byteArrays);
+        }
```

Comment:
It might be more work and lead to a bit of duplication, however, it would be great if we could push this method down to the implementation classes. The reason is that this avoid adding extra public API which might make harder to change things in the future.

Reply:
I've refactored this to be more centered around the JSONConverter. Please let me know whether this is still a concern.